index.js

```diff
@@ -44,6 +44,14 @@ var isArray = Array.isArray;
 // interdependence between different functions.
 var conjure;
 
+// a special hasOwnProperty function so we can override the built-in one
+
+var hasOwnProperty = Object.prototype.hasOwnProperty;
+
+function hop(obj, propName) {
```

Comment:
After talking to @AlmirKadric and him also being a bit confused at first on the `hop` naming, let me go and rename it :)